### ISSUE TITLE
Projection issues when converting meters to lat/lon for grid

### ISSUE THREAD

**Issue by RELNO** [CityScope/CSL_Hamburg_Grasbrook_Angular_Prototype#11]
Issue: 
Given mapbox's usage of [mercator](https://en.wikipedia.org/wiki/Mercator_projection#Web_Mercator) projections, deformation to objects converted from meters to lat/lon:
in 
```
      "longitude": 53.536898,
      "latitude": 10.007472,
```
<img width="385" alt="Screen Shot 2019-03-10 at 20 16 12" src="https://user-images.githubusercontent.com/14296774/54093795-cc550480-4371-11e9-8b24-2142e02177a3.png">

and in:

```
      "longitude": 3.536898,
      "latitude": 10.007472,
```
<img width="501" alt="Screen Shot 2019-03-10 at 20 16 35" src="https://user-images.githubusercontent.com/14296774/54093796-cced9b00-4371-11e9-86d7-e71906eb53b8.png">

This is a product of this method, which sees the lat/lon equally all over the mercator system, although the system stretches towards north/south [Hamburg].  

```
// Helper functions for working with latitude and longitude
  // issue with projection here:
  // https://github.com/mapbox/mapbox-gl-native/issues/11621
  metersOffsetToLatLong(
    centerLat: number,
    centerLon: number,
    dx: number,
    dy: number
  ): number[] {
    /**
     * Returns coordinate pair [newLat, newLon] where values are offset by
     * dx, dy meters from center lat, long.
     *
     * @param centerLat: central latitude point
     * @param centerLon: central longitude point
     * @param dx: x offset in meters
     * @param xy: y offset in meters
     * @returns array representing new [lat, lon] coodinate pair.
     **/
    let dLat = dx / EARTH_RADIUS;
    let dLon = dy / (EARTH_RADIUS * Math.cos((Math.PI * centerLat) / 180));
    dLat = (dLat * 180) / Math.PI;
    dLon = (dLon * 180) / Math.PI;
    let newLat = centerLat + dLat;
    let newLon = centerLon + dLon;
    return [newLat, newLon];
  }
```

Possible fix:
Find another method for drawing the grid cells which do not look at lat lon. 
Another option is to try and force localized projection, though this might be tricky. 

@agrignard can you take a look?

**Comment by RELNO** [comment]
similar discussion here: https://github.com/mapbox/mapbox-gl-native/issues/11621

**Comment by RELNO** [comment]
On a side note, this is also the reason the `rotation` method @aberke implemented works well near the equator:

<img width="833" alt="Screen Shot 2019-03-10 at 20 30 14" src="https://user-images.githubusercontent.com/14296774/54093952-72edd500-4373-11e9-9a6f-cea3c501719c.png">
[325 deg rotation] 

 But squeezes in the north.

**Comment by RELNO** [comment]
<img width="464" alt="Screen Shot 2019-03-11 at 20 42 07" src="https://user-images.githubusercontent.com/14296774/54167203-f9231d80-443e-11e9-9875-d846973858c3.png">

using https://github.com/proj4js/proj4js I could resolve it to one grid cell, though no rotation or translation yet:

-  WGS84 => Hamburg/North Germany Coor. system conversion
-  than added the distance for each edge [the grid cell size] 
- then convert BACK to WGS84 [Web mercator] and display 

This a long and cumbersome way, and I've kept very verbose in  https://github.com/CityScope/CSL_Hamburg_Grasbrook_Angular_Prototype/commit/3a2834b72ed46c06f597c6642808f663d8c7cae4 to allow comaring with current solution.   

@aberke would be great if you could take a look, tnx.

**Comment by aberke** [comment]
Hi @RELNO the discussion you linked to above (https://github.com/mapbox/mapbox-gl-native/issues/11621) resulted in success by using https://github.com/mapbox/sphericalmercator.  Did you try using that library as well?

**Comment by RELNO** [comment]
yeah, came across this discussion down the rabbit hole -- the bottomline was not promising though: 

>  I use a fixed zoom level of level 22 as its visual look is closest to what I need and also before. I went with visually look equal in size, and not pay attention much to physical size thus I did't limit anything.

level 22 is very far from what we're looking for -- proj4 https://proj4.org/ looks more promising for now

**Comment by agrignard** [comment]
Just to confirm I discussed with Gilles Gesquière from Lyon about this projection issue and he also confirmed that proj4 was one of the solution to this
